src/Core/AdminConsole/Services/Implementations/OrganizationService.cs

```diff
@@ -2027,15 +2052,6 @@ public async Task ValidateOrganizationUserUpdatePermissions(Guid organizationId,
         {
             throw new BadRequestException("Custom users can only grant the same custom permissions that they have.");
         }
-
-        // TODO: pass in the whole organization object when this is refactored into a command/query
-        // See AC-2036
-        var organizationAbility = await _applicationCacheService.GetOrganizationAbilityAsync(organizationId);
-        var flexibleCollectionsEnabled = organizationAbility?.FlexibleCollections ?? false;
-        if (flexibleCollectionsEnabled && newType == OrganizationUserType.Manager && oldType is not OrganizationUserType.Manager)
-        {
-            throw new BadRequestException("Manager role is deprecated after Flexible Collections.");
-        }
```

Comment:
This was in the wrong place - this method checks that the current user has permissions to invite/update the target user. Therefore it's only executed when there is a current user, which is not always the case (e.g. when updating via the public api)